src/adapt/viewer.js

```diff
@@ -608,6 +609,17 @@ adapt.viewer.Viewer.prototype.sizeIsGood = function() {
 
 /**
  * @private
+ * @param {{width: number, height: number}} pageSize
+ * @param {!Object<string, !{width: number, height: number}>} pageSheetSize
+ * @param {number} spineIndex
+ * @param {number} pageIndex
+ */
+adapt.viewer.Viewer.prototype.setPageSize = function(pageSize, pageSheetSize, spineIndex, pageIndex) {
+    this.pageSizes.push(pageSize);
```

Comment:
This method can be called multiple times for one page when cross references are resolved and re-layout is triggered.
Therefore the page size should be saved by specifying the page index as follows:
```javascript
    this.pageSizes[pageIndex] = pageSize;
```